### ISSUE TITLE
[charts] Allows to customize tradeof between page scroll and charts interaction on mobile

### ISSUE THREAD

**Issue by ventsarevich** [mui/mui-x#13885]
### Steps to reproduce

Check the video: https://github.com/user-attachments/assets/5317df9b-9669-43a0-ac94-ad97438730e5

There is no option to scroll horizontally or vertically over the chart on mobile devices.

Has some option to enable scroll over on mobiles?

### Current behavior

Can't scroll the page when touch the chart

### Expected behavior

Provide an ability to disable that behavior to have the ability to scroll over the large charts.

### Context

Check the video: https://github.com/user-attachments/assets/5317df9b-9669-43a0-ac94-ad97438730e5

There is no option to scroll horizontally or vertically over the chart on mobile devices. 
It's inconvenient for cases when charts is large and not scollable

### Your environment

System:
    OS: macOS 14.5
  Binaries:
    Node: 20.5.0 - ~/.nvm/versions/node/v20.5.0/bin/node
    npm: 9.8.1 - ~/WebstormProjects/Barrio/client/node_modules/.bin/npm
    pnpm: 9.4.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 126.0.6478.127
    Edge: Not Found
    Safari: 17.5
  npmPackages:
    @emotion/react: ^11.11.4 => 11.11.4 
    @emotion/styled: ^11.11.5 => 11.11.5 
    @mui/base:  5.0.0-beta.40 
    @mui/core-downloads-tracker:  5.15.18 
    @mui/icons-material: ^5.15.18 => 5.15.18 
    @mui/material: ^5.15.18 => 5.15.18 
    @mui/private-theming:  5.16.4 
    @mui/styled-engine:  5.16.4 
    @mui/system:  5.16.4 
    @mui/types:  7.2.15 
    @mui/utils:  5.16.4 
    @mui/x-charts: ^7.10.0 => 7.10.0 
    @mui/x-date-pickers: ^7.5.0 => 7.5.0 
    @types/react: 18.0.28 => 18.0.28 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: ~4.9.5 => 4.9.5 
    
 Browser: Safari (IOS)

**Search keywords**: mobile, scroll, chart

**Comment by ventsarevich** [comment]
This bug is relevant only to mobile devices. I reproduced this issue on IOS devices on Chrome and Safari and Android Chrome

**Comment by JCQuintas** [comment]
This was caused by a change to make users more easily see tooltips on mobile. We made it so you can drag your finger around the chart and the tooltip will show up as you drag. Maybe we need to think of an alternative... 🤔

**Comment by ventsarevich** [comment]
Having some flags to set the tooltip view option would be great. In some cases, showing a tooltip by dragging a finger over the chart could be convenient. But there are cases like a large chart or horizontal scroll when that behavior is just annoying and inconvenient.

**Comment by alexfauquette** [comment]
@ventsarevich Would removing `touch-action: none;` on the `<svg/>` solves your issue?

**Comment by michaelfaith84** [comment]
@alexfauquette That seems to have done it for me! Thanks!

**Comment by alexfauquette** [comment]
> but it does seem to break drag/long press.

You're talking about the tooltip behavior?

**Comment by colinnuk** [comment]
Before this change (https://github.com/mui/mui-x/pull/13692), I'd written a wrapper component which would intercept touchstart, touchmove & touchend events and send them to MUI as mouse events.

The component would only start sending the mouse events if a time threshold had passed (200ms). This meant that a user would 'touch and hold' and could then navigate the charts with the tooltip, or if they started scrolling (panning) then the mouse events would never fire, the tooltip wouldn't show up and the user would instead scroll the page. This just about worked ok - despite the hacky approach.

The PR linked above (while no doubt better for most users of this library) breaks this functionality completely.

The charts on my mobile page span the whole width of the page, so if I want the tooltip behaviour (which I do) - then the user is unable to scroll if they happen to touch a chart first, which isn't immediately obvious to them.

I think some form of threshold would be good - so if a user is long pressing they get the nice tooltip function, and if they only tap for a very short time then they can still scroll & pan the page.

**Comment by alexfauquette** [comment]
@colinnuk Thanks for the nicely detailed proposal. If I get it wrigh you propose to have timer starting at the touch down such that

- before 200ms (or other threshold) just let the mouse scroll
- after 200ms block scroll and started the chart interaction (tooltip and zoom)

Sounds like a nice behavior for the tooltip. I'm a bit more concerned by the pan for pro charts since those are also a scroll behavior

I'm also not sure about how to implement it, because it would require to remove the `touchAction: 'none'` leading to tons of edge cases where we need to take care of preventing default scroll

**Comment by colinnuk** [comment]
Yeah I don't see how it would be combined well with pan & zoom functionality.

After this change I was trying to ceate a wrapper which would change the touchAction from 'auto' to 'none' after a timeout, but it doesnt seem to work.

> I'm also not sure about how to implement it, because it would require to remove the `touchAction: 'none'` leading to tons of edge cases where we need to take care of preventing default scroll

I had to use `event.preventDefault()` for this to work.

In the meantime I think I am just going to implement a switch function to 'focus' on the graph with a 'close' button. I've seen this elsewhere (Garmin Connect app), where to get the detailed tooltip info you have to click on the graph which focuses the screen on that one graph and then provides a close/back button to go back to the main screen.

**Comment by alexfauquette** [comment]
Thanks for the detailed explanation. We are going to do more exploration on how other website are managing this tradeoff on this aspect to come with a robust solution.

**Comment by jisotalo** [comment]
I have multiple charts in one page and the user needs to scroll it using a touch screen. However, the cursor should be available as well (by double-tapping the chart first).

I had this issue fixed before as in comment above (https://github.com/mui/mui-x/issues/13885#issuecomment-2255819336). Today I noticed that it didn't work anymore.

The following seems to work:

```tsx
const [allowCursorOnMobile, setAllowCursorOnMobile] = useState(false);

//...

<LineChart
  /* other props removed */
  sx={{
    svg: {
      touchAction: allowCursorOnMobile ? 'none' : 'auto'
    }
  }}
/>
```

However at least on Chrome DevTools, the `auto` also causes issues with double-tapping (the browser zooms instead). Using `manipulation` seems to work..

```tsx
svg: {
  touchAction: allowCursorOnMobile ? 'none' : 'manipulation'
},
```

**Comment by RyanEwen** [comment]
I am struggling with the same issues - I need charts that don't interfere with scrolling, however I need to retain the ability to tap on a point and see a tooltip about the data at that point. I don't need any kind of touch-drag interaction on the graph in my case, but I do need to retain taps.

Is there a way to allow taps to show tooltips while not blocking scrolling?

**Comment by RyanEwen** [comment]
> In the meantime I think I am just going to implement a switch function to 'focus' on the graph with a 'close' button. 

Did you end up doing this? I think I would like to do the same and would love if you could share some code to help get me started more quickly 🙏

**Comment by JCQuintas** [comment]
> Is there a way to allow taps to show tooltips while not blocking scrolling?

Do you mean `tap` in the same way as click? That is, if the user taps down and up, the tooltip stays in the dom?

This is currently not possible.

If you mean `tap` as in "press", where if the user presses on the screen it displays the tooltip, then you can try our latest release.

We have recently implemented some improvements to the zoom&gesture handling in `v8.7.0`

**Comment by RyanEwen** [comment]
> Do you mean `tap` in the same way as click? That is, if the user taps down and up, the tooltip stays in the dom?
> 
> This is currently not possible.
 
I suppose I do - I didn't consider that the tooltip would just hide immediately. 

> If you mean `tap` as in "press", where if the user presses on the screen it displays the tooltip, then you can try our latest release.
> 
> We have recently implemented some improvements to the zoom&gesture handling in `v8.7.0`

I'm on 8.5.1. I'll try 8.7.0 and see if it helps. Thanks!

**Comment by RyanEwen** [comment]
I updated to latest (8.8.0) and now mobile users can scroll the page without the graphs preventing them from doing so while also being able to press on data points to see tooltips. Thank you!

**Comment by colinnuk** [comment]
> > In the meantime I think I am just going to implement a switch function to 'focus' on the graph with a 'close' button.
> 
> Did you end up doing this? I think I would like to do the same and would love if you could share some code to help get me started more quickly 🙏

@RyanEwen  Looks like you got it sorted with the latest MUI update? My workaround was working great on MUI v7 - but I stopped using it when I upgraded to v8.x

**Comment by RyanEwen** [comment]
> [@RyanEwen](https://github.com/RyanEwen) Looks like you got it sorted with the latest MUI update? My workaround was working great on MUI v7 - but I stopped using it when I upgraded to v8.x

Yes the changes in 8.7.0 were good enough so I no longer desire the functionality you described :)

**Comment by colinnuk** [comment]
Agreed, I'm just trying out 8.9 and it's much better. Good job!

**Comment by JCQuintas** [comment]
The general behavior here was updated and improved since this issue was opened, and it should not be causing problems for most users anymore.

Closing this one. If scrolling over a chart is still a problem on your setup, comment with the device and chart configuration and we can reopen.
